### PR TITLE
MGMT-14881 - return a valid error when no disks has been found

### PR DESCRIPTION
**Pull request by eifrach**
fix bug where host has failed validation when disk not been found.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [x] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/host/validator.go [hunk 1]
```diff
@@ -502,6 +506,13 @@ func (v *validator) hasMinValidDisks(c *validationContext) (ValidationStatus, st
 	if c.inventory == nil {
 		return ValidationPending, "Missing inventory"
 	}
+	inventory, err := c.inventoryCache.GetOrUnmarshal(c.host)
+	if err != nil {
+		return ValidationError, "Failed to load inventory"
+	}
+	if len(inventory.Disks) == 0 {
+		return ValidationError, "Failed Detected Disks"
```
Comment (avishayt) [inline internal/host/validator.go:514]:
```suggestion
		return ValidationError, "Failed to detect disks"
```

File: data/default_release_images.json [hunk 2]
```diff
@@ -32,17 +32,17 @@
         "cpu_architectures": [
             "x86_64"
         ],
-        "url": "quay.io/openshift-release-dev/ocp-release:4.11.46-x86_64",
-        "version": "4.11.46"
+        "url": "quay.io/openshift-release-dev/ocp-release:4.11.47-x86_64",
```
Comment (eliorerz) [inline data/default_release_images.json:35]: Why do you need to update that in this PR? It's better that this change will be changed as part of separate PR if it's not related to the problem that this PR fixs

**Review by eliorerz — CHANGES_REQUESTED** [review]

File: data/default_release_images.json (hunk 2 above)
Comment (eifrach) [reply data/default_release_images.json:35]: It's the same as [master](https://github.com/openshift/assisted-service/blob/master/data/default_release_images.json#L35)

Comment (eifrach) [reply data/default_release_images.json]: fixed

File: internal/host/validator.go [hunk 3]
```diff
@@ -128,8 +128,12 @@ func (c *validationContext) loadInventory() error {
...
 	}
-	if inventory.CPU == nil || inventory.Memory == nil || len(inventory.Disks) == 0 {
-		return errors.New("Inventory is not valid")
+	if inventory.Memory == nil {
+		return errors.New("Inventory is not valid, Memory not detected")
+	}
+
+	if inventory.CPU == nil {
+		return errors.New("Inventory is not valid, CPU not detected")
 	}
```
Comment (adriengentil) [inline internal/host/validator.go:137]: would it make sense to add a unit test for this change as well?

Comment (eifrach) [reply internal/host/validator.go:137]: Adding test for this is just adding an empty struct  to `inventory.Memory`  and `inventory.CPU`
we don't have any other / edge cases to this function. 

my guess seems not worth it.

**Review by adriengentil — APPROVED** [review]
